src/routes/docs/products/databases/atomic-numeric-operations/+page.markdoc

```diff
@@ -0,0 +1,524 @@
+---
+layout: article
+title: Atomic numeric operations
+description: Safely increment and decrement numeric fields without race conditions. Perfect for counters, quotas, inventory, and usage metrics in high-concurrency applications.
+---
+
+Atomic numeric operations allow you to safely increase or decrease numeric fields without fetching the full document. This eliminates race conditions and reduces bandwidth usage when updating any numeric values that need to be modified atomically, such as counters, scores, balances, and other fast-moving numeric data.
+
+# How atomic operations work {% #how-atomic-operations-work %}
+
+Instead of the traditional read-modify-write pattern, atomic numeric operations use dedicated methods to modify values directly on the server. The server applies the change atomically under concurrency control and returns the new value.
+
+**Traditional approach:**
+1. Fetch document → `{ likes: 42 }`
+2. Update client-side → `likes: 43`
+3. Write back → `{ likes: 43 }`
+
+**Atomic approach:**
+1. Call `incrementDocumentAttribute()` with the attribute name and the value to increment by
+2. Server applies atomically → `likes: 43`
+
+# When to use atomic operations {% #when-to-use-atomic-operations %}
+
+Atomic numeric operations work well for:
+
+- **Social features**: Likes, follows, comment counts
+- **Usage metering**: API credits, storage quotas, request limits
+- **Game state**: Scores, lives, currency, experience points
+- **E-commerce**: Stock counts, inventory levels
+- **Workflow tracking**: Retry counts, progress indicators
+- **Rate limiting**: Request counters, usage tracking
+
+# Perform atomic operations {% #perform-atomic-operations %}
+
+Use the `incrementDocumentAttribute` and `decrementDocumentAttribute` methods to perform atomic numeric operations. The server will apply these changes atomically under concurrency control.
+
+## Increment a field {% #increment-field %}
+
+{% multicode %}
+```client-web
+import { Client, Databases } from "appwrite";
+
+const client = new Client()
+    .setEndpoint('https://<REGION>.cloud.appwrite.io/v1') // Your API Endpoint
+    .setProject('<YOUR_PROJECT_ID>'); // Your project ID
+
+const databases = new Databases(client);
+
+const result = await databases.incrementDocumentAttribute(
+    '<DATABASE_ID>',
+    '<COLLECTION_ID>',
+    '<DOCUMENT_ID>',
+    'likes', // attribute
+    1 // value
+);
+
+console.log(result);
+```
+```client-flutter
+import 'package:appwrite/appwrite.dart';
+
+void main() async {
```

Comment:
Let's remove function wrappers, try/catches, and print statements from all examples, they distract from the important code